Refactor: Introduce AppSession and Remove Legacy SessionManager

MisbahAN opened the pull request:
This PR extracts only the AppSession-related changes from the previous (closed) full refactor PR.
It introduces a clean centralized AppSession for managing user session state and removes the legacy SessionManager.
No Firestore restructuring, no role-based DB abstractions, and no controller rewrites beyond replacing SessionManager usage.

MisbahAN committed:
Add global AppSession and remove obsolete SessionManager usage.

Orion-W commented on app/src/main/java/com/example/impact/view/AdminActivity.java:
```diff
@@ -35,9 +37,10 @@ public class AdminActivity extends AppCompatActivity {
-        adminId = getIntent().getStringExtra(LoginActivity.EXTRA_USER_ID);
+        adminId = AppSession.getUserId();
+        if (adminId == null) {
+            adminId = getIntent().getStringExtra(LoginActivity.EXTRA_USER_ID);
```
Can we put this logic in the AppSession class?

Orion-W commented on app/src/main/java/com/example/impact/view/LoginActivity.java:
```diff
@@ -127,9 +131,9 @@ private void navigateToRole(String role, String userId, @Nullable String email)
-        if (email != null) {
-            intent.putExtra("extra_user_email", email);
+        intent.putExtra(EXTRA_USER_ID, user.getId());
+        if (user.getEmail() != null) {
```
Do you think it is possible to handle most intent related operations in the AppSession class?

MisbahAN replied:
yeah that makes sense. i’ll move all the intent-related fallback logic into AppSession just like I did in the prev failed PR.

Orion-W reviewed (changes requested):
Looking good, thanks for getting back to me so quickly.
Do you think we can manage the "intent" related operations in the AppSession class? I see the following pattern a few times in this code:
```
thing = AppSession.getSomething();
if (thing == None):
    thing = intent.getSomething();
```
Instead I think it would be nice if when AppSession.getSomething() is called it checks if the value it should return exists and if not uses the intent to populate it before returning.
LMK if you think this is a bad idea.

MisbahAN committed:
Remove activity-level intent checks and hydrate session via AppSession.

MisbahAN commented:
@Orion-W I've pushed the requested update.

All intent-related fallback logic has now been moved into AppSession:

- Added `AppSession.setStartupIntent()` to capture the launch Intent
- Updated `getUserId()`, `getEmail()`, `getRole()`, and `getUser()` to hydrate themselves automatically from the Intent if in-memory values are null
- Removed all duplicated "if null, read from intent" patterns across `AdminActivity`, `OrganizerActivity`, `EntrantActivity`, `LoginActivity`, `SignupActivity`, and `SplashActivity`
- Activities now only call `AppSession.getX()`, keeping session access consistent and centralized

lmk if you'd like the hydration extended to any additional values, or if this structure should be adjusted further 👍

jeremycarefoot reviewed (approved):
Looks good to me

jeremycarefoot commented:
This PR has been floating for a few days so I am going to merge it.
